### PR TITLE
Add missing entry for 2.11 release notes

### PR DESCRIPTION
**Pull request by kotwanikunal**
### Description
- Adds in a missing entry for 2.11 release notes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch.release-notes-2.11.0.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### Added
 - Add coordinator level stats for search latency ([#8386](https://github.com/opensearch-project/OpenSearch/issues/8386))
 - Add metrics for thread_pool task wait time ([#9681](https://github.com/opensearch-project/OpenSearch/pull/9681))
+- Add parallel file and multi-stream download support for repository based replication ([#8596](https://github.com/opensearch-project/OpenSearch/pull/8596))
```
Comment (anasalkouz) [inline release-notes/opensearch.release-notes-2.11.0.md:8]: Is it only parallel file down support for now?

**Comment by noCharger** [comment]
Shall we say add meta entry in title?

**Review by reta — APPROVED** [review]

**Comment by andrross** [comment]
> Shall we say add meta entry in title?

I vote for the following:

```
Add parallel file download support for remote store based replication ([#8596](https://github.com/opensearch-project/OpenSearch/pull/8596))
```

**Comment by kotwanikunal** [comment]
@andrross / @anasalkouz Updated the entry.

**Review by anasalkouz — APPROVED** [review]

**Comment by kotwanikunal** [comment]
No code changes - release notes only. Pushing it through

**Pull request merged**